### PR TITLE
Support AM/PM in MSI token

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java [hunk 1]
```diff
@@ -53,29 +50,27 @@ public String getToken() {
...
-    }
-
     private static Long parseDateToEpochSeconds(String dateTime) {
         ClientLogger logger = new ClientLogger(MSIToken.class);
-        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss XXX");
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("M/d/yyyy H:mm:ss XXX", Locale.US);
+        // This is the format for app service on Windows as of API version 2017-09-01.
+        // The format is changed to Unix timestamp in 2019-08-01 but this API version
+        // has not been deployed to Linux app services.
+        DateTimeFormatter dtf_windows = DateTimeFormatter.ofPattern("M/d/yyyy K:mm:ss a XXX", Locale.US);
```
Comment (srnagar) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java]:
```suggestion
        DateTimeFormatter dtfWindows= DateTimeFormatter.ofPattern("M/d/yyyy K:mm:ss a XXX", Locale.US);
```

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java [hunk 2]
```diff
@@ -53,29 +50,27 @@ public String getToken() {
...
-    public boolean isExpired() {
-        OffsetDateTime now = OffsetDateTime.now();
-        OffsetDateTime expireOn = EPOCH.plusSeconds(parseDateToEpochSeconds(this.expiresOn));
-        return now.plusMinutes(5).isAfter(expireOn);
-    }
-
     private static Long parseDateToEpochSeconds(String dateTime) {
         ClientLogger logger = new ClientLogger(MSIToken.class);
-        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss XXX");
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("M/d/yyyy H:mm:ss XXX", Locale.US);
```
[the remark this answers is not in the document]
Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java]: I was thinking the service would always send US formatted datetimes, but i'm not sure

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java [hunk 3]
```diff
@@ -35,15 +36,11 @@
      * @param token the token string.
      * @param expiresOn the expiration time.
      */
-    public MSIToken(String token, OffsetDateTime expiresOn) {
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java]: Is this a Breaking change ?

Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java]: This is in the implmentation package so we should be fine

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java]: changed back to default which was working before

**Comment by jianghaolu** [comment]
Releasing now

**Pull request merged**